olp-cpp-sdk-core/tests/thread/ThreadPoolTaskSchedulerTest.cpp

```diff
@@ -280,3 +280,20 @@ TEST(ThreadPoolTaskSchedulerTest, Move) {
   thread_pool.reset();
   testing::Mock::VerifyAndClearExpectations(&mockop);
 }
+
+TEST(ThreadPoolTaskSchedulerTest, ScheduleTaskViaExecuteOrScheduleFunction) {
+  auto scheduler = std::make_shared<olp::thread::ThreadPoolTaskScheduler>(1);
+
+  std::promise<void> promise;
+  auto future = promise.get_future();
+  olp::thread::ExecuteOrSchedule(scheduler, [&]() { promise.set_value(); });
+
+  EXPECT_EQ(future.wait_for(std::chrono::milliseconds(kMaxWaitMs)),
+            std::future_status::ready);
```

Comment:
Instead of this, simply override the TaskScheduler with gmock, and check with EXPECT_CALL that the override is called or not.

Reply:
done